hack/generator/pkg/codegen/codegen_test.go

```diff
@@ -25,7 +27,73 @@ import (
 	"github.com/Azure/k8s-infra/hack/generator/pkg/jsonast"
 )
 
-func runGoldenTest(t *testing.T, path string) {
+type GoldenTestConfig struct {
+	HasArmResources      bool `json:"hasArmResources"`
+	InjectEmbeddedStruct bool `json:"injectEmbeddedStruct"`
+}
+
+func makeDefaultTestConfig() GoldenTestConfig {
+	return GoldenTestConfig{
+		HasArmResources:      false,
+		InjectEmbeddedStruct: false,
+	}
+}
+
+func loadTestConfig(path string) (GoldenTestConfig, error) {
+	result := makeDefaultTestConfig()
+
+	fileBytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		// If the file doesn't exist we just use the default
+		if os.IsNotExist(err) {
+			return result, nil
+		}
+
+		return result, err
+	}
+
+	err = json.Unmarshal(fileBytes, &result)
+	if err != nil {
+		return result, errors.Wrapf(err, "unmarshalling golden config %s", path)
+	}
+
+	return result, nil
+}
+
+func makeTestPackageReference() astmodel.PackageReference {
+	return astmodel.MakeLocalPackageReference("test", "v20200101")
+}
+
+func injectEmbeddedStructType() PipelineStage {
+	return MakePipelineStage(
+		"injectEmbeddedStructType",
+		"Injects an embedded struct into each object",
+		func(ctx context.Context, defs astmodel.Types) (astmodel.Types, error) {
+
+			results := make(astmodel.Types)
+			for _, def := range defs {
+				if _, ok := def.Type().(*astmodel.ObjectType); ok {
```

Comment:
Why not use the `IsObjectType()` predicate that's already defined in `object_type.go`?

Reply:
I forgot it existed =)

Will update